### PR TITLE
Remove naked gl from ApplicationOverlay and ApplicationCompositor

### PR DESCRIPTION
**Pull request by ZappoMan**
This is another step toward moving toward core profile and eliminating all the legacy OpenGL code.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 09430b6596107f9fbdf7dfd76ed6851dcc2411a7 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 069620b2d9cdfc7c840dbdae2a4df43b38fb2b8f available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 410e0a99ad58b8c2d26a8ac10432d42ab5038868 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

File: interface/src/ui/ApplicationCompositor.cpp [hunk 1]
```diff
@@ -209,9 +210,12 @@ void ApplicationCompositor::displayOverlayTexture(RenderArgs* renderArgs) {
     batch.setModelTransform(Transform());
     batch.setViewTransform(Transform());
     batch.setProjectionTransform(mat4());
-    batch._glBindTexture(GL_TEXTURE_2D, texture);
-    batch._glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-    batch._glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    //batch._glBindTexture(GL_TEXTURE_2D, texture);
+    //batch._glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    //batch._glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```
Comment (jherico) [inline interface/src/ui/ApplicationCompositor.cpp]: we'll need to test how this looks on HMD

File: interface/src/ui/ApplicationOverlay.cpp [hunk 2]
```diff
@@ -84,61 +78,92 @@ void ApplicationOverlay::renderOverlay(RenderArgs* renderArgs) {
...
+
+    // 2) clear it...
+    //glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    glm::vec4 color { 0.0f, 0.0f, 0.0f, 0.0f };
+    float depth = 0.0f;
+    int stencil = 0;
+    batch.clearFramebuffer(gpu::Framebuffer::BUFFER_COLORS | gpu::Framebuffer::BUFFER_DEPTH, color, depth, stencil);
+    //batch.clearColorFramebuffer(_overlayFramebuffer->getBufferMask(), glm::vec4(0.0f, 0.0f, 0.0f, 0.0f));
+
+    int width = _overlayFramebuffer ? _overlayFramebuffer->getWidth() : 0;
```
Comment (jherico) [inline interface/src/ui/ApplicationOverlay.cpp]: if overlayframebuffer is null, we should bail out early.

File: interface/src/ui/ApplicationOverlay.cpp [hunk 3]
```diff
@@ -84,61 +78,92 @@ void ApplicationOverlay::renderOverlay(RenderArgs* renderArgs) {
...
+    renderOverlays(renderArgs); // renders Scripts Overlay and AudioScope
+    renderStatsAndLogs(renderArgs);  // currently renders nothing
+    renderDomainConnectionStatusBorder(renderArgs); // renders the connected domain line
+    renderQmlUi(renderArgs); // renders a unit quad with the QML UI texture
+
+    //_overlayFramebuffer->release(); // now we're done for later composition
+    //batch.setFramebuffer(nullptr);
+
+    renderArgs->_context->syncCache();
+    renderArgs->_context->render(batch);
```
Comment (jherico) [inline interface/src/ui/ApplicationOverlay.cpp]: I think we need to reset renderArgs->_batch to null at this point.

**Comment by ZappoMan** [comment]
@jherico - any idea why this PR isn't working? namely on the mac it produces garbage framebuffers as if the clearFramebuffer() call isn't working... I haven't tested it on windows yet, maybe it works there... have you done anything yet with the gpu::Framebuffer classes?

File: interface/src/ui/ApplicationCompositor.cpp (hunk 1 above)
Comment (ZappoMan) [inline interface/src/ui/ApplicationCompositor.cpp]: According to @samcake we set up the "samplers" when we set up the textures in the frame buffers. He assures me that the gpu::sampler classes are as flexible as the standard openGL versions, we just need to make sure we pick the right options. Once I get it rendering correctly I'll definitely be testing in the hmd.

File: interface/src/ui/ApplicationOverlay.cpp (hunk 2 above)
Comment (ZappoMan) [inline interface/src/ui/ApplicationOverlay.cpp]: good point

File: interface/src/ui/ApplicationOverlay.cpp (hunk 3 above)
Comment (ZappoMan) [inline interface/src/ui/ApplicationOverlay.cpp]: good idea. although it seems like the _batch gets reset later on in displaySide()

**Comment by hifi-gustavo** [comment]
Build console output for commit: efa2001325980420ca89897a84e232bfa067386d available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 33464dfd6c5d27fdb22959da619c5c3caedadef3 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4dadf60724c57bc60b411fcd23569da8edf55779 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ccd2f98a76db96ffd53e02786ad80e181efec0fa available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 03a44a1c80cf7e816201f02944d3a585dd3b295e available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

File: interface/src/ui/ApplicationCompositor.cpp (hunk 1 above)
Comment (ZappoMan) [inline interface/src/ui/ApplicationCompositor.cpp]: Note, I tested it, it looks good... as in same as production.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c20c6e4b90081b4b36310e830fcb3789678ce2af available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8692f51399ca69cbb0302509c14c10359b4b4e82 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 695470b1c1bf7ba7a1ebd7ab0d7812841ab1fd5f available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eb989fd7250d690f1c92c821416c230d4f6a696a available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26e8090ec1f2f8813ce93f6a5d55f12799b6b302 available at: https://gist.github.com/1680e6609ea9193750ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5352.dmg

**Pull request merged**